sys/include/net/gnrc/nettype.h

```diff
@@ -77,7 +77,8 @@ typedef enum {
 #ifdef MODULE_GNRC_TCP
     GNRC_NETTYPE_TCP,           /**< Protocol is TCP */
 #endif
-#ifdef MODULE_GNRC_UDP
+#if defined(MODULE_GNRC_UDP) || \
+    (defined(MODULE_GNRC_SIXLOWPAN_IPHC_NHC) && defined(MODULE_GNRC_SIXLOWPAN_ND_BORDER_ROUTER))
```

Comment:
Make it just `MODULE_GNRC_SIXLOWPAN_IPHC_NHC`. There is no need for the relation to `gnrc_udp` since no functions from that module is used. If we want to make UDP support for NHC optional we should introduce a pseudo module `gnrc_sixlowpan_iphc_nhc_udp`